src/trace.ml

```diff
@@ -551,19 +565,20 @@ let put_alloc s now ~length ~nsamples ~is_major
 
 let get_alloc ~parse_backtraces evcode cache alloc_id b =
   let open Read in
-  let is_short, length, nsamples, is_major =
+  let is_short, length, nsamples, source =
     match evcode with
     | Ev_short_alloc n ->
-       true, n, 1, false
+       true, n, 1, Allocation_source.Minor
     | Ev_alloc -> begin
        let length = get_vint b in
        let nsamples = get_vint b in
-       let is_major =
+       let source : Allocation_source.t =
          match get_8 b with
-         | 0 -> false
-         | 1 -> true
+         | 0 -> Minor
+         | 1 -> Major
+         | 2 -> External
          | _ -> bad_format "is_major" in
```

Comment:
Should this now say `bad_format "source"`?

Reply:
Fixed